src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1277,9 +1277,15 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
                 }
             }
 
-            // Add the content headers
-            if (request.Content == null)
+            // For other methods like Put where empty content has meaning, we need to fill in the content
+            if (request.Content is null)
             {
+                // If this is a Get request and there is no content, then don't fill in the content as empty content gets rejected by some web services
+                if (request.Method == HttpMethod.Get && ContentType is null)
+                {
+                    return;
```

Comment:
Why do we ignore Header below? I think we need the headers.

Also if method is GET ContentType must be null https://stackoverflow.com/questions/5661596/do-i-need-a-content-type-header-for-http-get-requests
I'd expect this should be fixed on top of the method.

Reply:
Per your link, a `Get` without a body should NOT have a ContentType.  So in this case, if the user did not specify an explicit ContentType, that header isn't being added.

Reply:
I mean there can be other headers.

Reply:
I see what you are saying, but it doesn't apply here.  The loop below this is specifically for Content headers.  If the Content is null, then there are no headers.

Reply:
Yes, but I did not find any differences in the code (in pwsh and .Net) for content headers and non-content headers.
So I'm worried about losing some headers like authorization.

Reply:
Other headers are handled elsewhere, this is specifically regarding Content headers